Dockerfile

```diff
@@ -241,6 +241,26 @@ VOLUME /var/lib/docker
 # Wrap all commands in the "docker-in-docker" script to allow nested containers
 ENTRYPOINT ["hack/dind"]
 
+FROM dev AS cross
+RUN dpkg --add-architecture armhf
+RUN dpkg --add-architecture arm64
+RUN dpkg --add-architecture armel
+# These crossbuild packages rely on gcc-<arch>, but this doesn't want to install
+# on non-amd64 systems.
+# Additionally, the crossbuild-amd64 is currently only on debian:buster, so
+# other architectures cannnot crossbuild amd64.
+RUN if [ "$(go env GOHOSTARCH)" = "amd64" ]; then \
+	apt-get update \
+	&& apt-get install -y \
+		crossbuild-essential-armhf \
+		crossbuild-essential-arm64 \
+		crossbuild-essential-armel \
+		libseccomp-dev:armhf \
+		libseccomp-dev:arm64 \
+		libseccomp-dev:armel; \
+	fi
+COPY . /go/src/github.com/docker/docker
```

Comment:
what if we use `if/else` logic from https://medium.com/@tonistiigi/advanced-multi-stage-build-patterns-6f741b852fae in here and build with `docker build --build-arg CROSS=true .` if you want extra packages

Reply:
Yes, and actually could be used to compile runc/containerd/etc.

Reply:
Updated this so that the Makefile has a `DOCKER_CROSS` var that it will pass along to the build.
DOCKER_CROSS is set to true when DOCKER_CROSSPLATFORMS is set OR the `cross` target is used, otherwise it just passes in whatever value from the user.